src/journal/views.py

```diff
@@ -1392,36 +1392,73 @@ def search(request):
     """
     articles = []
     search_term = None
+    keyword = None
+    if request.POST and 'clear' in request.POST:
+        return logic.unset_search_GET_variables(request)
+        # return logic.unset_search_session_variables(request)
 
-    if request.POST:
-        search_term = request.POST.get('search')
-        request.session['article_search'] = search_term
-        return redirect(reverse('search'))
+    search_term, keyword, sort, redir = logic.handle_search_controls(request)
 
-    if request.session.get('article_search'):
-        search_term = request.session.get('article_search')
+    if redir:
+        return redir
+    #if no order, set order to title (alphabetical). this happens when receiveing a GET request for keyword.
+    if request.GET and not sort:
+        sort='title'
 
+    if search_term:
+        # return two lists of pks based on search term.
         article_search = submission_models.Article.objects.filter(
             (Q(title__icontains=search_term) |
-             Q(subtitle__icontains=search_term)) &
-            Q(journal=request.journal)
+             Q(keywords__word__icontains=search_term) |
+             Q(subtitle__icontains=search_term)), 
+            journal=request.journal, 
+            stage=submission_models.STAGE_PUBLISHED,
+            date_published__lte=timezone.now()
         )
-        article_search = [article for article in article_search]
+        article_search_unsorted = [article.pk for article in article_search]
 
-        author_search = search_term.split(' ')
```

Comment:
Dropping this line means that when I search for "Mauro Sanchez" my own articles won't show up (and I bet you authors will get frustrated if their own articles don't show up in the search :grimacing: )

Reply:
I had broken this up because if you don't search for someone's name **exactly** they won't show up because it was a first_name= and last_name=. 
```
>>> author_search = ['Burkhardt']
>>> FrozenAuthor.objects.filter(
... (Q(first_name__in=author_search) |
... Q(last_name__in=author_search))
... )
<QuerySet []>

>>> author_search = ['richard', 'Burkhardt']
>>> FrozenAuthor.objects.filter(
... (Q(first_name__in=author_search) |
... Q(last_name__in=author_search))
... )
<QuerySet []>

>>> author_search = ['Richard', 'Burkhardt']
>>> FrozenAuthor.objects.filter(
... (Q(first_name__in=author_search) |
... Q(last_name__in=author_search))
... )
<QuerySet [<FrozenAuthor: Richard W. Burkhardt, Jr.>]>
```
This allowed for one to be able to search for partial names and was not case sensitive... 
Combining the two is a possibility:
```
search_term = 'rich'
author_search = ['Richard', 'Burkhardt']
FrozenAuthor.objects.filter(
(Q(first_name__icontains=search_term) |
Q(last_name__icontains=search_term))
|
(Q(first_name__in=author_search) |
Q(last_name__in=author_search)) 
)
```
But I am not sure how much of a load this sort of a query would be for a large database system. It's a lot of text searching. Especially if the queries are combined like you suggest in your next comment below.

Reply:
Yes, I think combining the two approaches is the right solution as the query is going to do a full scan on the table anyways. Perhaps a sensible restriction here would be to limit the search term to a minimum of 3 characters